### DISCUSSION TITLE
What to do about the lab app?

### DISCUSSION THREAD

**Discussion by srallen** [zooniverse/front-end-monorepo#2430] [General]
https://github.com/zooniverse/pfe-lab

The lab app was a very early experiment on trying to break up the monolith of PFE. It was so early that work on it was prior to some of our later tech decisions. We'll need to do more work on it soon, but we should make some maintenance decisions about it first:

- Do we move the existing lab app into the monorepo then attempt to update it to our newer setup using grommet, mobx, etc?
- Do we basically start over with a brand new next.js app and all our current tech stack?
  - If we start over, what is salvage from the current lab app?
  - Is there an argument for using other tech choices and if so, then what?

**Comment by rogerhutchings** [comment]
Thoughts:

- These pages aren't going to be indexable, so SEO isn't a consideration in whether to do SSR or not.
- I like the idea of using our established tech stack, I don't want another thing for people to have to learn.
- I _suspect_ trying to move over existing code will be more trouble than it's worth; I'd suggest starting off with a fresh app and copying bits in as and when possible.

More as things come to mind...

**Comment by srallen** [comment]
Most of the app are forms. We know that we want to move toward more explicit displays of state and requiring users to hit 'save' or 'submit'. I can see the prevalent use of forms as an argument to not do a SPA. However, we have a nice setup now with Grommet and the theme, so that should help cut down on development time and wheel reinvention. Grommet's team has been recommending [formik](https://github.com/jaredpalmer/formik). Perhaps we should come up with a small trial of it to see if it'll work for us?
